src/shared/lua/Buildables.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+===========================================================================
+
+Unvanquished GPL Source Code
+Copyright (C) 2024 Unvanquished Developers
+
+This file is part of the Unvanquished GPL Source Code (Unvanquished Source Code).
+
+Unvanquished Source Code is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Unvanquished Source Code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Unvanquished Source Code.  If not, see <http://www.gnu.org/licenses/>.
+
+In addition, the Unvanquished Source Code is also subject to certain additional terms.
+You should have received a copy of these additional terms immediately following the
+terms and conditions of the GNU General Public License which accompanied the Unvanquished
+Source Code.  If not, please request a copy in writing from id Software at the address
+below.
+
+If you have questions concerning this license or the applicable additional terms, you
+may contact in writing id Software LLC, c/o ZeniMax Media Inc., Suite 120, Rockville,
+Maryland 20850 USA.
+
+===========================================================================
+*/
+#ifdef BUILD_CGAME
+
+#include "shared/lua/Buildables.h"
+
+namespace Shared {
+namespace Lua {
+
+
+#define GETTER(name) { #name, Get##name }
+
+BuildableProxy::BuildableProxy( int buildable ) :
+	attributes( BG_Buildable( buildable ) ) {}
+
+#define GET_FUNC( var, type ) \
+static int Get##var( lua_State* L ) \
+{ \
+	BuildableProxy* proxy = LuaLib<BuildableProxy>::check( L, 1 ); \
+	lua_push##type( L, proxy->attributes->var ); \
+	return 1; \
+}
+
+#define GET_FUNC2( name, var, type ) \
+static int Get##name( lua_State* L ) \
+{ \
+	BuildableProxy* proxy = LuaLib<BuildableProxy>::check( L, 1 ); \
+	lua_push##type( L, var ); \
+	return 1; \
+}
+
+GET_FUNC2( name, proxy->attributes->humanName, string )
+GET_FUNC( info, string )
+GET_FUNC( icon, string )
+GET_FUNC2( build_points, proxy->attributes->buildPoints, integer )
+GET_FUNC2( unlock_threshold, proxy->attributes->unlockThreshold, integer )
+GET_FUNC( health, integer )
+GET_FUNC2( regen_rate, proxy->attributes->regenRate, integer )
+GET_FUNC2( splash_damage, proxy->attributes->splashDamage, integer )
```

Comment:
splash_damage and splash_radius should have more informative names indicating that they are for the death explosion, e.g. `death_splash_damage`.

Reply:
I'd prefer to keep the lua names the same as the C++ names